packages/state-transition/src/block/processWithdrawals.ts

```diff
@@ -154,13 +154,10 @@ export function getExpectedWithdrawals(
   for (n = 0; n < bound; n++) {
     // Get next validator in turn
     const validatorIndex = (nextWithdrawalValidatorIndex + n) % validators.length;
-    const partiallyWithdrawnBalance = withdrawals
-      .filter((withdrawal) => withdrawal.validatorIndex === validatorIndex)
-      .reduce((acc, withdrawal) => acc + Number(withdrawal.amount), 0);
 
     const validator = validators.getReadonly(validatorIndex);
     const balance = isPostElectra
-      ? balances.get(validatorIndex) - partiallyWithdrawnBalance
+      ? balances.get(validatorIndex) - getPartiallyWithdrawnBalance(withdrawals, validatorIndex)
```

Comment:
I think as an improvement in the future, we can avoid looping through `withdrawals` in each sweep by constructing a map `Map<ValidatorIndex, number>` to cache partial withdrawn balance by validator index.